registration/registration_test.go

```diff
@@ -349,8 +358,51 @@ func Test_CheckCertificate(t *testing.T) {
 	t.Run("certification check enabled", func(t *testing.T) {
 		pub, private, err := ed25519.GenerateKey(nil)
 		require.NoError(t, err)
+
+		expiration := time.Now().Add(time.Hour)
+		data, err := shared.EncodeCert(&shared.Cert{Pubkey: nodeID, Expiration: &expiration})
+		require.NoError(t, err)
+
+		defaultValidCert := &shared.OpaqueCert{
+			Data:      data,
+			Signature: ed25519.Sign(private, data),
+		}
+		pubKeyHint := pub[:shared.CertPubkeyHintSize]
+
 		powVerifier := mocks.NewMockPowVerifier(gomock.NewController(t))
 
+		const keysNum = 3
+
+		dir := t.TempDir()
+		t.Cleanup(func() {
+			os.RemoveAll(dir)
+		})
+
+		var (
+			trustedKeyCert *shared.OpaqueCert
+			trustedKey     []byte
+		)
+
+		for i := 0; i < keysNum; i++ {
+			encodedPubkey, private, err := generateTestKeyPair()
+			require.NoError(t, err)
+
+			createTestKeyFile(t, dir, fmt.Sprintf("valid_key_%d.key", i), encodedPubkey)
+
+			if i == keysNum-1 {
+				expiration := time.Now().Add(time.Hour)
+				data, err := shared.EncodeCert(&shared.Cert{Pubkey: nodeID, Expiration: &expiration})
+				require.NoError(t, err)
+
+				trustedKeyCert = &shared.OpaqueCert{
+					Data:      data,
+					Signature: ed25519.Sign(private, data),
+				}
+
+				trustedKey = encodedPubkey
+			}
+		}
```

Comment:
NIT: Can be simplified

```suggestion
		// generate keys
		for i := range keysNum - 1 {
			pubKey, _, err := ed25519.GenerateKey(nil)
			require.NoError(t, err)

			path := filepath.Join(dir, fmt.Sprintf("valid_key_%d.key", i))
			require.NoError(t, os.WriteFile(path, []byte(base64.StdEncoding.EncodeToString(pubKey)), 0o644))
		}
		// generate key and certificate
		trustedKey, private, err = ed25519.GenerateKey(nil)
		require.NoError(t, err)

		path := filepath.Join(dir, "valid_key.key")
		require.NoError(t, os.WriteFile(path, []byte(base64.StdEncoding.EncodeToString(trustedKey)), 0o644))
		expiration = time.Now().Add(time.Hour)
		data, err = shared.EncodeCert(&shared.Cert{Pubkey: nodeID, Expiration: &expiration})
		require.NoError(t, err)

		trustedKeyCert = &shared.OpaqueCert{
			Data:      data,
			Signature: ed25519.Sign(private, data),
		}
```